app/Http/Controllers/Api/ExperienceSkillsController.php

```diff
@@ -48,4 +49,70 @@ public function destroy(ExperienceSkill $experienceSkill)
     {
         $experienceSkill->delete();
     }
+
+    public function batchStore(BatchStoreExperienceSkill $request)
+    {
+        $newExperienceSkills = $request->validated();
+        $response = [];
+
+        foreach ($newExperienceSkills as $newExperienceSkill) {
+            // Restore soft deleted experienceSkill if it exists, otherwise create a new one.
+            $softDeletedExperienceSkill = ExperienceSkill::withTrashed()
+                ->where([
+                ['skill_id', $newExperienceSkill['skill_id']],
+                ['experience_id', $newExperienceSkill['experience_id']],
+                ['experience_type', $newExperienceSkill['experience_type']]
+            ])->first();
+            if ($softDeletedExperienceSkill) {
+                $softDeletedExperienceSkill->restore();
+                if ($newExperienceSkill['justification'] !== null && $newExperienceSkill['justification'] !== '') {
+                    $softDeletedExperienceSkill->justification = $newExperienceSkill['justification'];
+                }
+                $softDeletedExperienceSkill->save();
+                array_push($response, $softDeletedExperienceSkill);
+            } else {
+                $experienceSkill = new ExperienceSkill($newExperienceSkill);
+                $experienceSkill->skill_id = $newExperienceSkill['skill_id'];
+                $experienceSkill->experience_id = $newExperienceSkill['experience_id'];
+                $experienceSkill->experience_type = $newExperienceSkill['experience_type'];
+                $experienceSkill->justification = $newExperienceSkill['justification'];
+                $experienceSkill->save();
+                $experienceSkill->fresh();
+                array_push($response, $experienceSkill);
+            }
+        }
+
+        return new JsonResource($response);
```

Comment:
I also tend to use `return JsonResource::collection($response);` when it's an array of items rather than a single object being returned. Not sure if it actually makes a difference but it's [outlined in the docs here](https://laravel.com/docs/8.x/eloquent-resources#resource-collections).